src/compositor/seat/touch.c

```diff
@@ -87,7 +87,7 @@ wlc_touch_touch(struct wlc_touch *touch, uint32_t time, enum wlc_touch_type type
          break;
 
          case WLC_TOUCH_MOTION:
-            wl_touch_send_motion(wr, time, slot, wl_fixed_from_double(pos->x), wl_fixed_from_double(pos->y));
+            wl_touch_send_motion(wr, time, slot, pos->x, pos->y);
             break;
```

Comment:
wl_fixed isn't int32_t either, though you should change wl_fixed_from_double to wl_fixed_from_int here, it's binary scaled unit (truncated using division and expanded using multiplication, don't ask the wl people were on acid when they did it and how they implemented it is even more crazy)

That implicit double conversion most likely is just copy paste from pointer.c that does use double for x and y, unlike here.